db/sql/030_create_billable.sql

```diff
@@ -0,0 +1,124 @@
+DROP MATERIALIZED VIEW IF EXISTS billable;
+CREATE MATERIALIZED VIEW IF NOT EXISTS billable AS with
+
+-- extract useful stuff from usage events
+-- we treat both apps and services as billable "resources" so normalize the fields
+-- we normalize states to just STARTED/STOPPED because we treat consecutive STARTED to mean "update"
+events as (
```

Comment:
`events` doesn't seem to select `previous_instance_count` or `previous_memory_in_mb_per_instance`, so how does it produce data we can calculate scaling events from?

Reply:
Everything is operating on the idea that we have "good" data.

We can make that assumption because we are going generate all the start events using either the `RepairEvents` method or by call the "purge and reset events" cloud controller equivalent.

So we are not "guessing" previous states. Usage is calculated from a `START` event to the next `START` or `STOP` event.

Reply:
So, if in the last 5 minutes of the calendar month I scale my up then down again I only get billed for 5 minutes?

Reply:
No. 

I think maybe you are framing the problem in terms of "months", and this is causing some confusion.

We have access to ALL of the events from the moment we started collecting them, and we are working under the assumption that all this data is "good".

So in the scenario you describe, we may not have seen a `START` event during _this_ month, but we will have seen one before that. So when a query is made that requests the billable resources limited to a slice a time, ie between t1 and t2 it will take that into account.

Does that make sense?

Reply:
ah, awesome, thanks.